pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -234,6 +236,20 @@ func (asw *actualStateOfWorld) MarkVolumeAsDetached(
 	asw.DeleteVolumeNode(volumeName, nodeName)
 }
 
+func (asw *actualStateOfWorld) MarkDesireToDetachVolume(
+	volumeName api.UniqueVolumeName, nodeName string) error {
+	asw.Lock()
+	defer asw.Unlock()
+	return asw.removeVolumeFromReportAsAttached(volumeName, nodeName)
+}
+
+func (asw *actualStateOfWorld) UnmarkDesireToDetachVolume(
+	volumeName api.UniqueVolumeName, nodeName string) {
+	asw.Lock()
+	defer asw.Unlock()
+	asw.addVolumeToReportAsAttached(volumeName, nodeName)
+}
+
 func (asw *actualStateOfWorld) AddVolumeNode(
```

Comment:
Update the comments on the `AddVolumeNode(...)` interface method and `DetachRequestedTime` field (old comment said this method will reset the detach time which is no longer the case).